### DISCUSSION TITLE
Download more then one model at a time

### DISCUSSION THREAD

**Discussion by ctrlcmdshft** [Avnsx/fansly#45] [Ideas]
can i just download all models consecutively or list more then one manually.  If i sub to 5 models and want to run one command to grab them all is that possible

**Comment by NumNuts101** [comment]
This would be nice

**Comment by ooono-developer** [comment]
I totally agree

**Comment by Avnsx** [comment]
I mean this would be nice to have, but the code base is too entangled to introduce this feature at the moment.

**Comment by ctrlcmdshft** [comment]
What about showing you a list of models you sub too and then you can choose which via number

**Comment by ZilockeT** [comment]
You could also use something like AutoIt as a frontend, when you run it have it read a list of models from a text file, ask which model you want to download and then write it into the config file.

I used to have a script like this for selecting private WoW servers from a list. Not really too hard.

Could *possibly* even automate it, although I'd hesitate to do that without experimenting with how Fansly would react to someone suddenly using a lot of bandwidth over a list of models.

If you do sub to quite a few, though, maybe download 1-2 a day, then just rotate through your list to update them every so often.

**Comment by prof79** [comment]
I wanted that, too - even thought of contributing/taking a look at the repo. But sorry, appreciating all your achievements, but the code is a monolithic, unstructured mess - main script flow intertwined with helper function definitions and so on. If there was a "download_user(name, path)" function one could simply loop over it but not in that case. Config parts validation, user validation, download of parts and so on could and should have been pulled out to separate functions/modules.
In the current state it will be hard to find helpers/contributers unless you permitted a major rewrite.

**Reply by Avnsx** [reply]
I do permit a major re-write feel free to open a pull request with your changes.

**Reply by prof79** [reply]
That's great, gimme some time - currently laden with tasks and work, might be able to start getting into it in around 1-1,5 weeks at the earliest.
